### PR TITLE
Fix #1957. Boards dev guide.

### LINKED ISSUE #1957: Write Boards plugin developers guide
## Summary

The [contributors / developer setup](https://www.focalboard.com/contribute/getting-started/) and [readme](https://github.com/mattermost/focalboard#readme) currently have instructions to develop against Personal Server. We should write a guide to setup a development environment for the Mattermost Boards plugin.

Looking for someone to write the initial draft, to be reviewed by the dev team. This should include best-practices for a developer workflow to build/test the plugin locally.

More and more features are specific to Boards, and require testing the plugin.

### PR DESCRIPTION
**Pull request by chenilim**
#### Summary
Draft of developers guide for building and deploying the Boards plugin.

I based most of this from the [thread here](https://community.mattermost.com/core/pl/649rnxgkijfdux5xsmio348gar). Should be a starting point for adding more tips and tricks, e.g. around debugging.

#### Ticket Link
#1957

### CODE REVIEW

File: docs/mattermost-boards-dev-guide.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+3. Enable it (if needed)
+
+Alternatively, you can install Mattermost locally and use `make deploy` to auto-deploy it for you:
+
+First, build and run Mattermost locally:
+1. Follow the [Mattermost Developers Guide](https://developers.mattermost.com/contribute/server/developer-setup/) to set up your environment
+  * In particuler, make sure Docker is set up and running
+2. Clone / fork [mattermost-webapp](https://github.com/mattermost/mattermost-webapp) and `make build`
+3. Clone / fork [mattermost-server](https://github.com/mattermost/mattermost-server)
+4. Edit `config.json` with a site url, enable [plugin uploads](https://developers.mattermost.com/integrate/admin-guide/admin-plugins-beta/#custom-plugins), and enable [local mode](https://docs.mattermost.com/configure/configuration-settings.html#enable-local-mode)
```
Comment (chenilim) [inline docs/mattermost-boards-dev-guide.md:50]: @sbishel, @wiggin77, do you know where the `config.json` file is in the mattermost-server repo? I see a `test-config.json` file, is that the one?

Comment (wiggin77) [reply docs/mattermost-boards-dev-guide.md:50]: @chenilim it is generated on first run.

Comment (chenilim) [reply docs/mattermost-boards-dev-guide.md:50]: Thanks! I also found a [thread on this today](https://community-daily.mattermost.com/private-core/pl/n9qqdgg9ptnppfjrmcri1bze8y). So, should a new developer run `make config-reset`, edit the generated config.json, then run `make run-server`?

File: docs/mattermost-boards-dev-guide.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Alternatively, you can install Mattermost locally and use `make deploy` to auto-deploy it for you:
+
+First, build and run Mattermost locally:
+1. Follow the [Mattermost Developers Guide](https://developers.mattermost.com/contribute/server/developer-setup/) to set up your environment
+  * In particuler, make sure Docker is set up and running
+2. Fork [mattermost-webapp](https://github.com/mattermost/mattermost-webapp), clone it locally, and `make build`
+3. Fork [mattermost-server](https://github.com/mattermost/mattermost-server) and clone it locally
+3. Run `make config-reset` to generate the `config/config.json` file
+4. Edit `config/config.json`:
+  * Set `SiteURL` to `http://localhost:8065` ([docs](https://docs.mattermost.com/configure/configuration-settings.html#site-url))
```
Comment (chenilim) [inline docs/mattermost-boards-dev-guide.md:52]: Is the the correct setting to use for local debugging?

File: docs/mattermost-boards-dev-guide.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Now, to build and deploy the Boards plugin:
+1. Clone / fork [mattermost/focalboard](https://github.com/mattermost/focalboard)
+2. Install the dependencies (see above)
+3. Run:
+```
+cd mattermost-plugin
+run make webapp
+run make deploy
+```
```
Comment (jespino) [inline docs/mattermost-boards-dev-guide.md:66]: Remember to add here your own Tips and Tricks in the comments.

**Comment by chenilim** [comment]
Ok, I've manually run through these steps on an Ubuntu EC2 server, and was able to complete them, EXCEPT for one thing:

When running `make run-server` under `mattermost-server`, I get an error:
```
Failed to start up plugins","error":"mkdir ./client/plugins: no such file or directory"
```
And `make deploy` fails because "plugins are not enabled".

The workaround was to manually `mkdir client` then `make stop-server` and `make run-server`, then deploy again.

Did anyone else run into this? Do you have a `client` folder under `mattermost-server`?

**Review by harshilsharma63 — APPROVED** [review]

**Comment by harshilsharma63** [comment]
> Ok, I've manually run through these steps on an Ubuntu EC2 server, and was able to complete them, EXCEPT for one thing:
> 
> When running `make run-server` under `mattermost-server`, I get an error:
> 
> ```
> Failed to start up plugins","error":"mkdir ./client/plugins: no such file or directory"
> ```
> 
> And `make deploy` fails because "plugins are not enabled".
> 
> The workaround was to manually `mkdir client` then `make stop-server` and `make run-server`, then deploy again.
> 
> Did anyone else run into this? Do you have a `client` folder under `mattermost-server`?

@chenilim I didn't face this issue when setting up a clean environment on MacOS.

Also, the plugin development steps for Focalboard worked perfectly fine.

**Comment by chenilim** [comment]
Thanks @harshilsharma63! I'll check this in. Please feel free to submit additional PRs to add more tips and tricks (e.g. for debugging).

**Pull request merged**